methods.py

```diff
@@ -730,7 +733,10 @@ def add_shared_library(env, name, sources, **args):
 
 
 def add_library(env, name, sources, **args):
-    library = env.Library(name, sources, **args)
+    if not env["shared"] or str(env.Flatten(sources)[0]).endswith(env["OBJSUFFIX"]):
```

Comment:
this  `endswith` condition is not super robust and I don't like it. This is kind of a temporary hack to support static building of the few libraries I mentioned had complications switching to shared. If every library can be built shared then this can go away. If this is unacceptable I can either tackle libraries that are having complications or come up with a more robust way to identify if a library is needing to build static.

Reply:
all libraries are built shared now so this hack goes away!